### DISCUSSION TITLE
gmod-minify

### DISCUSSION THREAD

**Discussion by giscus** [AMD-NICK/blog.amd-nick.me#25] [Blog comments]
# gmod-minify

code minification banner

https://blog.amd-nick.me/gmod-minify

**Comment by CMAULTOP** [comment]
Можно даже интереснее сделать, через Github Actions автоматическая обфускация кода, а уже на сервере будет подгружаться только сборка обфусцированная.

**Reply by AMD-NICK** [reply]
Да, я хотел написать об этом в разделе "У меня хостинг", но поленился. Если найдутся желающие -- допишу

**Comment by Portik123** [comment]
А где ты вводил cd /home/user и так далее, просто не могу понять

**Reply by AMD-NICK** [reply]
Твой вопрос свидетельствует о том, что ты совсем не имеешь опыта работы с командной строкой. Тут тебе сначала нужно подучить азы, потом можно перечитывать эту статью

**Comment by SkrideMat** [comment]
Иногда ломает lua скрипты

**Reply by SkrideMat** [reply]
Временно решил проблему так
```
find "$BASEPATH/$serv/garrysmod/addons" -name '*.lua' \( -name 'cl_*.lua' -o -name '*client*.lua' \) -exec bash -c 'lua minify.lua minify "$1" > tmp.lua || cp -v "$1" tmp.lua ; cp -v tmp.lua "$1"' sh {} \;
```

**Reply by AMD-NICK** [reply]
А что значит ломает и в чем решение?

**Reply by SkrideMat** [reply]
После минификации кода, некоторые скрипты начинают выдавать ошибки. Я не выяснил, что именно ломается, в каждом из них что-то свое (кто-то жалуется на переменные, кто-то на функции). Решение - минификация клиентских скриптов

**Reply by AMD-NICK** [reply]
Странное поведение. Интересно узнать причину. Если будешь разбираться - дай знать

**Comment by brawillion** [comment]
Так код в мунифицированном виде все еще выполняет функции. Файл может быть использован как как скрипт внутри он полюбому запустит
